Catalog/src/Application/Common/Behaviours/LoggingBehaviour.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Catalog.Application.Common.Interfaces;
+using MediatR.Pipeline;
+using Microsoft.Extensions.Logging;
+
+namespace Catalog.Application.Common.Behaviours;
+
+public class LoggingBehaviour<TRequest> : IRequestPreProcessor<TRequest> where TRequest : notnull
```

Comment:
Although microservices architecture recommends copying code rather then reusing code, common code can still be placed and shared between microservices as nuget. Implementing logger or any similar behaviour service in each service is wasteful.